### PR TITLE
Re-export `pending` from `std` rather than redefining it

### PR DESCRIPTION
**Pull request by joshtriplett**
This avoids having two versions of `pending` in programs that use
both futures_lite and the version from std.

### CODE REVIEW

**Comment by taiki-e** [comment]
Thanks for the PR. Unfortunately, if the user implements the same trait on both types, this is a breaking change.

(https://github.com/smol-rs/futures-lite/issues/35 is an issue to track this change.)

File: src/future.rs [hunk 1]
```diff
@@ -110,45 +109,7 @@ pub fn block_on<T>(future: impl Future<Output = T>) -> T {
...
-}
-
-impl<T> Future for Pending<T> {
-    type Output = T;
-
-    fn poll(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<T> {
-        Poll::Pending
-    }
-}
+pub use std::future::{pending, Pending};
```
Comment (zseri) [inline src/future.rs:112]:
```suggestion
pub use core::future::{pending, Pending};
```

**Comment by zseri** [comment]
Given that this is a breaking change anyways, it doesn't really make sense to merge this because the user would need to manually fix their code anyways (given a corresponding bump of the crate version). So I'd reject this PR.

**Comment by notgull** [comment]
> Given that this is a breaking change anyways, it doesn't really make sense to merge this because the user would need to manually fix their code anyways (given a corresponding bump of the crate version). So I'd reject this PR.

I'd agree with rejection.

**Pull request closed without merging**